stdlib/socket/src/main/ballerina/socket/client_socket.bal

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents socket client and related actions.
+#
+# + remotePort - the remote port number to which this socket is connected
+# + localPort - the local port number to which this socket is bound
+# + remoteAddress - the remote IP address string in textual presentation to which the socket is connected
+# + localAddress - the local IP address string in textual presentation to which the socket is bound
+public type Client client object {
+
+    private ClientConfig config;
+    public int remotePort = 0;
+    public int localPort = 0;
+    public string? remoteAddress = ();
+    public string? localAddress = ();
+
+    public function __init(ClientConfig? clientConfig) {
+        if (clientConfig is ClientConfig) {
+            self.config = clientConfig;
+            self.initEndpoint(clientConfig);
+            self.start();
+        }
+    }
+
+    extern function initEndpoint(ClientConfig clientConfig);
+
+    extern function start();
+
+    # Write given data to the client socket.
```

Comment:
Recommended way of writing docs is to phrase the description in form of (Writes, Closes etc..) instead of (Write, Close etc..). Shall we fix all the docs in this pattern?